### ISSUE TITLE
Exchanges referencing "US" zone no longer appear on map

### ISSUE THREAD

**Issue by daveachuk** [tmrowco/electricitymap#1403]
Change #1366 added a new zone which covers Washington State in the US under the Bonneville Power Administration, and I think since then the CA-BC->US exchange that used to show up on the map no longer does.

It looks like [config/exchanges.json](https://github.com/tmrowco/electricitymap/blob/master/config/exchanges.json) includes a few references to the "US" zone which no longer exists anywhere on the map, so that's probably why. Maybe.

Namely: 

- CA-BC->US  (should be changed to US-BPA?)
- CA-AB->US (should be changed to US-MT?)
- US->US-CA (Not sure about this one, the CSV imported in the parser contains only aggregate imports, not broken down by connection)

**Comment by jarek** [comment]
Possible reason is relation to #1391 which made internal back-end changes to how zones with no parsers are handled. Namibia is also impacted by this.

**Comment by corradio** [comment]
I think this is related to the changes we did with @maxbellec.
I won't have time to fix this weekend, but we'll look at that asap.
No data loss is involved, so that's a good start.

**Comment by systemcatch** [comment]
@daveachuk you're right the first 2 exchanges should be CA-BC->US-BPA and CA-AB->US-MT, do you want to correct them or shall I?

For US->US-CA that's an aggregate of all exchanges to CAISO which we just show as a generic arrow.

**Comment by jarek** [comment]
The exact details of these western Canadian exchanges are a bit of a meaningless difference because we know nothing about Montana, don't have real time data for BC, and the situation of BPA is not all that clear (http://explorer.watttime.org shows significant parts of Washington, possibly including the BC exchanges, as being outside the BPA area)

**Comment by systemcatch** [comment]
I think BC does have exchanges with BPA (https://www.bchydro.com/content/BCHydro/en/energy-in-bc/operations/transmission/transmission-system/actual-flow-data.html?WT.ac=sia_transmission_actualflowdata) mentions them by name, but you're right that this is mostly insignificant since there is no data.

**Comment by maxbellec** [comment]
actually this seems to be a front-end error, not a back-end one: exchanges are in the states. @jbuverud @corradio do you know what can be the source of this?
![screenshot at 2018-05-15 11-58-19](https://user-images.githubusercontent.com/7253902/40050294-5278f6f6-5837-11e8-8fe5-00a2d0331878.png)
![screenshot at 2018-05-15 11-59-33](https://user-images.githubusercontent.com/7253902/40050343-7354f9c4-5837-11e8-98df-5301f82281da.png)

**Comment by jbuverud** [comment]
You're right @maxbellec, I think this is due to us now removing exchange information from countries with no data (like @jarek said). I'll change the implementation and see if I can get this problem fixed.

**Comment by jarek** [comment]
This is still an issue. The change referenced has been deployed. Is there more to the problem?

**Comment by maxbellec** [comment]
the website is only update when we update the backend. We will tonight

**Comment by corradio** [comment]
@maxbellec Actually I deployed it this morning: https://tmrow-team.slack.com/archives/C2DQPTHA6/p1526459994000026

**Comment by corradio** [comment]
Re-opening as issue is still happening

**Comment by jbuverud** [comment]
Which exchanges are not showing, exactly?

**Comment by corradio** [comment]
CA-US and Namibia Zambia

**Comment by corradio** [comment]
Actually @maxbellec it seems like this is a backend issue, as the `netFlow` shows `undefined`:

![image](https://user-images.githubusercontent.com/1655848/40140559-2e6cbfee-594b-11e8-9e3e-0d35669e05b2.png)

**Comment by systemcatch** [comment]
Is it something to with if an exchange is flowing from a zone that has no parsers/data to a zone with data it is getting ignored? This is the case with the Namibian exchanges right now. CA-BC->US-BPA is flowing from BPA at the moment and showing on the map so that one is fixed in one direction.

**Comment by maxbellec** [comment]
yes it's probably something like that @systemcatch, though I haven't yet figured out where we do that. Thanks for investigating

**Comment by systemcatch** [comment]
You found the source of the problem @maxbellec?

**Comment by maxbellec** [comment]
yeah, I hadn't coded in Javascript for a while and forgot that `someObject[keyDoesntExist]` is `undefined`, and does not throw an error :rage:
